### ISSUE TITLE
Individual Prject. Checkpoint 2.

### ISSUE THREAD

**Issue by tvdmitrii** [tvdmitrii/courseSchedulePlanner-api#5]
@pawaitemadisoncollege checkpoint 2 is complete.

- [x] Database designed and created.
> Database design is complete. Although it will likely require a few tweaks here and there to work with Cognito and to accommodate some nuances that I will discover once working on the UI.
- [x] At least one DAO with full CRUD (create, read, update, delete) implemented with Hibernate.
> DAO is fully implemented, but I might discover that I need more methods as I am working on the WebApp.
- [x] DAO is fully unit tested.
- [x] Log4J is implemented (no System.out.printlns)

It feels like the major systems like database, REST API, and WebApp are in place and deployed and things will go smoother from now on. Now I can work from UI and build down endpoints, DTOs, and methods. I have added the pages and user stories I will work on to the plan.

The project was split up into three repositories to house [REST API + database](https://github.com/tvdmitrii/courseSchedulePlanner-api) code, [WebApp](https://github.com/tvdmitrii/courseSchedulePlanner-webApp), and the [REST API interface](https://github.com/tvdmitrii/courseSchedulePlanner-interface) that they share. It would probably be better to further separate REST API and database, but I might come back to it at the end if I have time.

**Comment by pawaitemadisoncollege** [comment]
Hi @tvdmitrii! You've made some good early progress with your indie project, and I agree with your decision to split up the application into multiple repositories. You've included this information clearly in the readme.md with links to the other repos which will make it easy for anyone reviewing this to find all the relevant parts. 

**Database designed and created** 

In reviewing the ERD, the database appears to be well-designed.

- [ ] One thing - the user table will need to store the unique identifier from cognito - if that is email, then your table is all set. If something else, like a username, then that column will need to be added. I'm certain from your comments that this is on your radar already. 

**DAO**

Implemented beyond expectations for this point in the semester.  Multiple entities with relationships are well-tested to cover a variety of scenarios. Well done! 

**Log4J is implemented (no System.out.printlns)**

Logging present. No printlns found when searching the repos. :) 

**Time log and Plan**

This is up-to-date through week 7 (with items checked off in the plan).

- [ ] It's nearing the time to add more specific tasks for the upcoming weeks, 11-14. This would be a good goal to have ready for  checkpoint 3.

**Checkpoint 2: met** 
**Week 8- restful service consumption: met**

**Comment by tvdmitrii** [comment]
Thank you for your feedback @pawaitemadisoncollege!

1. I have added a column to user table to store UUID that Cognito uses to identify the users. I decided to remove all other user fields like email, username, name. First, I do not actually need those. They will come from Cognito and will be stored in session. Second, I was a bit concerned about all the user data in the database being easily accessible through REST API, and I do not have time to implement proper authentication for it. I will only have `id`, `uuid`, and `role` that will dictate whether the user has admin privileges.
2. I have updated my project plan to specify when I will be working on each of the remaining user stories.
